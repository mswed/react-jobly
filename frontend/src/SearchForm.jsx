import { useState } from 'react';
import { Form, Button, InputGroup } from 'react-bootstrap';
const SearchForm = ({ search }) => {
  const INITIAL_STATE = '';
  const [searchField, setSearchField] = useState(INITIAL_STATE);
  const handleChange = (evt) => {
    setSearchField(evt.target.value);
  };
  const handleSubmit = (evt) => {
    evt.preventDefault();
    search(searchField);
    setSearchField(INITIAL_STATE);
  };
  return (
    <Form onSubmit={handleSubmit}>
      <InputGroup className="m-3">
        <Form.Control placeholder="Enter search term..." onChange={handleChange} value={searchField} />
        <Button variant="primary" type="submit">
          Search!
        </Button>
      </InputGroup>
    </Form>
  );
};

export default SearchForm;
