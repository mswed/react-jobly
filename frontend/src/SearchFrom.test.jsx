import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import SearchForm from './SearchForm';

it('renders without crashing', async () => {
  render(<SearchForm />);
});

it('matches snapshot', async () => {
  const { asFragment } = render(<SearchForm />);

  expect(asFragment()).toMatchSnapshot();
});
