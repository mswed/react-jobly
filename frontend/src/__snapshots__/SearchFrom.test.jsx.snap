// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="my-3 input-group"
    >
      <input
        class="form-control"
        placeholder="Enter search term..."
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Search!
      </button>
    </div>
  </form>
</DocumentFragment>
`;
